.github/workflows/ci.yaml

```
@@ -3,120 +3,95 @@
+      - '.gitignore'
+      - '.jujuignore'
+      - 'LICENSE'
+      - '**.md'
+      - .github/renovate.json5
   schedule:
-    - cron: "53 0 * * *" # Daily at 00:53 UTC
+    - cron: '53 0 * * *' # Daily at 00:53 UTC
   # Triggered on push to branch "main" by .github/workflows/release.yaml
   workflow_call:
 
 jobs:
   lint:
     name: Lint
+    uses: canonical/data-platform-workflows/.github/workflows/lint.yaml@v13.1.2
+
+  lib-check:
+    name: Check libraries
     runs-on: ubuntu-latest
     timeout-minutes: 5
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - name: Install tox
-        # TODO: Consider replacing with custom image on self-hosted runner OR pinning version
-        run: python3 -m pip install tox
-      - name: Run linters
-        run: tox run -e lint
+        with:
+          fetch-depth: 0
+      - run: |
+          # Workaround for https://github.com/canonical/charmcraft/issues/1389#issuecomment-1880921728
+          touch requirements.txt
+      - name: Check libs
+        uses: canonical/charming-actions/check-libraries@2.4.0
+        with:
+          credentials: ${{ secrets.CHARMHUB_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          use-labels: false
+          fail-build: ${{ github.event_name == 'pull_request' }}
 
   unit-test:
-    name: Unit tests
+    name: Unit test charm
     runs-on: ubuntu-latest
     timeout-minutes: 25 # TODO: reduce to 5 minutes
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
-      - name: Install tox
-        # TODO: Consider replacing with custom image on self-hosted runner OR pinning version
-        run: python3 -m pip install tox
+        uses: actions/checkout@v4
+      - name: Install tox & poetry
+        run: |
+          pipx install tox
+          pipx install poetry
       - name: Run tests
         run: tox run -e unit
 
   build:
+    name: Build "${{ matrix.path }}" charm
+    uses: canonical/data-platform-workflows/.github/workflows/build_charm.yaml@v13.1.2
     strategy:
       fail-fast: true
       matrix:
-        charms: [".", "tests/integration/ha_tests/application_charm", "tests/integration/relation_tests/application-charm"]
-    name: Build ${{matrix.charms}} charm
-    uses: canonical/data-platform-workflows/.github/workflows/build_charm_without_cache.yaml@v8
+        path:
+          - "."
+          - "tests/integration/ha_tests/application_charm"
+          - "tests/integration/relation_tests/application-charm"
     with:
-      charmcraft-snap-revision: 1349
-      path-to-charm-directory: ${{matrix.charms}}
+      cache: true
+      path-to-charm-directory: "${{ matrix.path }}"
 
   integration-test:
-    strategy:
-      fail-fast: false
-      matrix:
-        tox-environments:
-          - charm-integration
-          - ha-integration
-          - tls-integration
-          - metrics-integration
-    name: ${{ matrix.tox-environments }}
+    name: Integration test charm | 3.1.7
```

```suggestion
    name: Integration test charm
```
nit: no need to include version since you're only testing on one juju version; also don't want to have another place where version needs to be updated if changed (that could get out of sync & provide incorrect info)

---

I set this one so it's clear to us what juju version is used, to remind us to upgrade/bump if needed

---

(we have a renovate config for that)

at least, I would encourage you to consider putting it in a variable so that it correctly reflects whichever arg is passed/version is actually used

---

Added a variable for it

---

nit: ^